File: Vanilla/sources/zdoc/media/lua/shared/Library/PZCalendar.lua
---@class PZCalendar : zombie.util.PZCalendar
---@field private calendar Calendar
PZCalendar = {}

---@public
---@return PZCalendar
function PZCalendar:getInstance() end

---@public
---@return long
function PZCalendar:getTimeInMillis() end

---@public
---@param arg0 int
---@param arg1 int
---@param arg2 int
---@param arg3 int
---@param arg4 int
---@return void
function PZCalendar:set(arg0, arg1, arg2, arg3, arg4) end

---@public
---@param arg0 int
---@return boolean
function PZCalendar:isLeapYear(arg0) end

---@public
---@return Date
function PZCalendar:getTime() end

---@public
---@param arg0 int
---@return int
function PZCalendar:get(arg0) end

---@public
---@param arg0 long
---@return void
function PZCalendar:setTimeInMillis(arg0) end

File: Vanilla/sources/zomboid/zombie/util/PZCalendar.java
package zombie.util;

import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;
import java.util.Objects;

public final class PZCalendar {
   private final Calendar calendar;

   public static PZCalendar getInstance() {
      return new PZCalendar(Calendar.getInstance());
   }

   public PZCalendar(Calendar var1) {
      Objects.requireNonNull(var1);
      this.calendar = var1;
   }

   public void set(int var1, int var2, int var3, int var4, int var5) {
      this.calendar.set(var1, var2, var3, var4, var5);
   }

   public void setTimeInMillis(long var1) {
      this.calendar.setTimeInMillis(var1);
   }

   public int get(int var1) {
      return this.calendar.get(var1);
   }

   public final Date getTime() {
      return this.calendar.getTime();
   }

   public long getTimeInMillis() {
      return this.calendar.getTimeInMillis();
   }

   public boolean isLeapYear(int var1) {
      return ((GregorianCalendar)this.calendar).isLeapYear(var1);
   }
}
